app/products/calls/hooks.test.ts

```diff
@@ -0,0 +1,271 @@
+        });
+
+        it('shows alert when calls not enabled', async () => {
+            mockClient.getEnabled.mockResolvedValue(false);
+            const mockAlert = jest.spyOn(Alert, 'alert');
+            const {result} = renderHook(() => useTryCallsFunction(mockFn));
+            const [tryFn] = result.current;
+
+            await act(async () => {
+                await tryFn();
+            });
+
+            expect(mockFn).not.toHaveBeenCalled();
+            expect(mockAlert).toHaveBeenCalled();
+        });
+    });
+
+    describe('usePermissionsChecker', () => {
+        beforeEach(() => {
+            jest.spyOn(AppState, 'addEventListener');
+        });
+
+        it('checks permissions when not granted', async () => {
+            const mockCheck = jest.spyOn(Permissions, 'check').mockResolvedValue(Permissions.RESULTS.GRANTED);
+            const {result} = renderHook(() => usePermissionsChecker(false));
+
+            expect(result.current).toBe(false);
+
+            // Simulate app becoming active
+            await act(async () => {
+                AppState.currentState = 'active';
+                const callback = (AppState.addEventListener as jest.Mock).mock.calls[0][1];
+                await callback('active');
+            });
+
+            expect(mockCheck).toHaveBeenCalled();
+            expect(initializeVoiceTrack).toHaveBeenCalled();
+            expect(setMicPermissionsGranted).toHaveBeenCalledWith(true);
+            expect(result.current).toBe(true);
+        });
+
+        it('skips check when already granted', () => {
+            const mockCheck = jest.spyOn(Permissions, 'check');
+            renderHook(() => usePermissionsChecker(true));
+            expect(mockCheck).not.toHaveBeenCalled();
+        });
+    });
+
+    describe('useCallsAdjustment', () => {
+        beforeEach(() => {
+            (useIncomingCalls as jest.Mock).mockReturnValue({incomingCalls: []});
+            (useChannelsWithCalls as jest.Mock).mockReturnValue({});
+            (useCallsState as jest.Mock).mockReturnValue({calls: {}, myUserId: 'user1'});
+            (useGlobalCallsState as jest.Mock).mockReturnValue({micPermissionsGranted: true});
+            (useCurrentCall as jest.Mock).mockReturnValue(null);
+            (queryAllActiveServers as jest.Mock).mockReturnValue({
+                fetch: () => Promise.resolve([{id: 'server1'}]),
+            });
+        });
+
+        it('calculates adjustment with no calls', () => {
+            const {result} = renderHook(() => useCallsAdjustment('server1', 'channel1'));
+            expect(result.current).toBe(0);
+        });
+
+        it('includes current call bar height', () => {
+            (useCurrentCall as jest.Mock).mockReturnValue({id: 'call1', channelId: 'channel1'});
+            const {result} = renderHook(() => useCallsAdjustment('server1', 'channel1'));
+            expect(result.current).toBe(68); // CURRENT_CALL_BAR_HEIGHT + 8 padding
```

Comment:
nit -- maybe this should be the constant + padding, incase (probably: when) the constant gets changed by design in the future?